firmware/arduino/chessboard/circleFunction.ino

```diff
@@ -0,0 +1,224 @@
+int pulsesPerSlope[NUM_CIRCLES][NUM_SLOPES_PER_QUARTER_CIRCLE];
+
+enum SlopeToIndex
+{
+    SLOPE_HORIZONTAL = 0,
+    SLOPE_EIGHTH = 1,
+    SLOPE_QUARTER = 2,
+    SLOPE_HALF = 3,
+    SLOPE_ONE = 4,
+    SLOPE_TWO = 5,
+    SLOPE_FOUR = 6,
+    SLOPE_EIGHT = 7,
+    SLOPE_VERTICAL = 8
+};
+
+
+// Makes a full circle of the given size (0=largest, NUM_CIRCLES-1=smallest) 
+// starting from the given quarter (0=top, 1=left, 2=bottom, 3=right).
+// calculatePulsesPerSlope() must be called before makeCircle()
+void makeCircle(int circle, int firstQuarter)
+{
+    bool firstPass = true;
+    int slopeIndex;
+
+    setScale(xMotor, EIGHTH_STEPS);
+    digitalWrite(xMotor[STEP_PIN], LOW);
+
+    for (int quarterCircle = firstQuarter; quarterCircle != firstQuarter || firstPass; quarterCircle = (quarterCircle + 1) % 4)
+    {
+        firstPass = false;
+
+        // Set Y-direction
+        if (quarterCircle == 0 || quarterCircle == 1)
+            digitalWrite(yMotor[DIR_PIN], DOWN);
+        else
+            digitalWrite(yMotor[DIR_PIN], UP);
+
+        // Set X-direction
+        if (quarterCircle == 0 || quarterCircle == 3)
+            digitalWrite(xMotor[DIR_PIN], LEFT);
+        else
+            digitalWrite(xMotor[DIR_PIN], RIGHT);
+
+        // Set starting point for the while loop
+        if (quarterCircle == 0 || quarterCircle == 2)
+            slopeIndex = SLOPE_HORIZONTAL;
+        else
+            slopeIndex = NUM_SLOPES_PER_QUARTER_CIRCLE - 1;
+
+        // Iterate through pulsesPerSlope
+        while (slopeIndex >= SLOPE_HORIZONTAL  &&  slopeIndex <= SLOPE_VERTICAL)
+        {
+            // Y-scale
+            if (slopeIndex == SLOPE_TWO)
```

Comment:
I would double check to see if there's a way to improve this. For example, SLOPE_TWO triggers the y scale to be 1/4, and the x scale to be 1/8. This does equal a slope of two, but we could use higher scales to do it faster. For example, a y scale of 1 and an x scale of 1/2.

Reply:
Doing it like this allows you to reduce lines by using EIGHTH_STEPS by default, but it makes the travel speeds longer.